XenAdmin/Diagnostics/Checks/UpgradeRequiresEUA.cs

```diff
@@ -90,15 +93,31 @@ private void FetchHostEua(Host host)
                 _euas.Add(eua);
             }
         }
+
         protected override Problem RunCheck()
         {
+            if (Hosts.Count == 0)
+            {
+                return null;
+            }
+
             foreach (var host in Hosts)
             {
                 var hotfix = HotfixFactory.Hotfix(host);
                 if (hotfix != null && hotfix.ShouldBeAppliedTo(host))
                     return new HostDoesNotHaveHotfixWarning(this, host);
             }
 
+            string upgradePlatformVersion = null;
+            if (_installMethodConfig != null)
+                Host.TryGetUpgradeVersion(Hosts.FirstOrDefault(), _installMethodConfig, out upgradePlatformVersion, out _);
+
+            // There's no EUA for Pre-82X versions
+            if (!Helpers.Post82X(upgradePlatformVersion))

```

Comment:
Is it checking source XS version or destination XS version? The test failed on RPU from Yangtze to Master, will this use case return null here?

Reply:
This new check is for the destination XS. 

The test failed from Yangtze to Master because XenCenter could not fetch the EUA (was not present), that's not a XenCenter issue.

This is addressing upgrades to non-Master. e.g. Stockolhm to Yangtze.

I do need to make a further tweak to support this, however. Will send in a new change

Reply:
True but we should still add a version check, and while we're at it prevent EUA from showing when we know it's not there, anyway